crates/turbopack-convert-trace/src/main.rs

```diff
@@ -411,6 +448,92 @@ fn main() {
             match task {
                 Task::Enter { id, root } => {
                     let mut span = take(&mut spans[id]);
+                    let mut count = span.count;
+                    let mut items = take(&mut span.items);
+                    if graph && !items.is_empty() {
+                        let parent_name = &*span.name;
+                        let mut groups = IndexMap::new();
+                        let mut self_items = Vec::new();
+                        fn add_items_to_groups<'a>(
+                            groups: &mut IndexMap<Cow<'a, str>, Vec<SpanItem>>,
+                            self_items: &mut Vec<SpanItem>,
+                            spans: &mut Vec<Span<'a>>,
+                            parent_count: &mut u32,
+                            parent_name: &str,
+                            items: Vec<SpanItem>,
+                        ) {
+                            for item in items {
+                                match item {
+                                    SpanItem::SelfTime { .. } => {
+                                        self_items.push(item);
+                                    }
+                                    SpanItem::Child(id) => {
+                                        let key = spans[id].name.clone();
+                                        if key == parent_name {
```

Comment:
Does this also imply `id == parent_id`? Or does each span have a unique id regardless of the called function?

Reply:
The same function can be called multiple times (different ids) and it would have the same name.